src/Core/Framework/DataAbstractionLayer/Command/RefreshIndexCommand.php

```diff
@@ -51,6 +52,7 @@ protected function configure(): void
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $this->io = new ShopwareStyle($input, $output);
+        $this->noProgress = (bool) $input->getOption('no-progress');
```

Comment:
Can we just assign `null` to `$this->io` when the `no-progress` option is true? Then we can remove the trait changes

Reply:
In this case, it would only work because there are no other outputs than the progress bar. As soon as another output message gets added (like a success-message) the setting of `$this->io` to `null` would also the output of that message. `$this->noProgress` is more fine-grained in this regard.  

For example, let's say I would add the `no-progress` to the `CartMigrateCommand` via setting of `$this->io` to `null` which also uses the `ConsoleProgressTrait`: The success message would not be printed.

Reply:
@shyim what do you think? I would rather null the io and if some other command wants to write other messages they can just create a local io or refactor.

Reply:
i would also say local io would be better

Reply:
@M-arcus ping me when you refactored and we can get this merged :)

Reply:
@AydinHassan ping :smiley: